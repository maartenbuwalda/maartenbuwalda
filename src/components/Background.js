import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import Fade from '@material-ui/core/Fade'

const Wrapper = styled.div`
  background-color: ${props => props.color.hex.colorMutedRegular};
  background-image: url(${props => props.image});
  background-repeat: no-repeat;
  background-position: center center;
  min-height: 100vh;
  width: 100vw;
  background-size: cover;
`

const Background = ({ image, color, children }) => (
  <Fade in={true} timeout={500}>
    <Wrapper color={color} image={image}>
      {children}
    </Wrapper>
  </Fade>
)

Background.propTypes = {
  image: PropTypes.string,
  color: PropTypes.object,
  children: PropTypes.node,
}

export default Background
