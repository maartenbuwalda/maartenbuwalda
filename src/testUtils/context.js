const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
