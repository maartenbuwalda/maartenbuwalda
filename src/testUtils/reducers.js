export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
