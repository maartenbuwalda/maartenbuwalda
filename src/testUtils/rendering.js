import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}

export const connectedComponentRenders = component => {
  const middlewares = []
  const mockStore = configureStore(middlewares)
  const initialState = {}
  const store = mockStore(initialState)
  
  it('renders correctly', () => {
    const shallowed = shallow(component, store)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}
