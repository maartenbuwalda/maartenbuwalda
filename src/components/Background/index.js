import Background from './Background'
import { withRouter } from 'react-router'

export default withRouter(Background)
