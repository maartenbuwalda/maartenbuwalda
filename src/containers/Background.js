import Background from '../components/Background'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { getDominantColor } from '../actions/color'

export const mapStateToProps = state => {
  return {
    color: state.color,
  }
}

export const mapDispatchToProps = dispatch => {
  return {
    getDominantColor: image => {
      dispatch(getDominantColor(image))
    },
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Background))
