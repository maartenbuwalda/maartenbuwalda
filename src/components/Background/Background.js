import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { ColorContext } from '../../context/color'

const BackgroundDarkOverlay = styled.div`
  position: fixed;
  z-index: -1;
  top: 0;
  min-height: 100vh;
  min-width: 100vw;
  background: linear-gradient(
    0deg, rgba(${({ theme }) => theme.rgb.colorMutedDark}, 0.1), rgba(${({ theme }) => theme.rgb.colorMutedDark}, 0)
  );
`

const ColoredBackground = styled.div`
  transition: background-color 1s;
  position: relative;
  min-height: 100vh;
  background-color: rgba(${({ theme }) => theme.rgb.colorMutedDark}, 0.1);
`

const Background = ({ children }) => (
  <ColorContext.Consumer>
    {({ theme }) => (
      <ColoredBackground theme={theme}>
        {children}
        <BackgroundDarkOverlay theme={theme} />
      </ColoredBackground>
    )}
  </ColorContext.Consumer>
)

Background.propTypes = {
  children: PropTypes.node,
  color: PropTypes.object,
}

export default Background
