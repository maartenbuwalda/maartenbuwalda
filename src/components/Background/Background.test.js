import React from 'react'
import Background from './index'
import { componentRenders } from '../../testUtils/rendering'

describe('Background component', () => {
  const component = <Background>Children</Background>
  componentRenders(component)
})
