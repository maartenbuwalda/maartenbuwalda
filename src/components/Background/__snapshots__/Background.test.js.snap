// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background component mounts correctly 1`] = `
.c1 {
  position: fixed;
  z-index: -1;
  top: 0;
  min-height: 100vh;
  min-width: 100vw;
  background: linear-gradient( 0deg,rgba(,0.1),rgba(,0) );
}

.c0 {
  -webkit-transition: background-color 1s;
  transition: background-color 1s;
  position: relative;
  min-height: 100vh;
  background-color: rgba(,0.1);
}

<Background>
  <styled.div
    theme={
      Object {
        "hex": Object {
          "colorDark": "",
          "colorLight": "",
          "colorMutedDark": "",
          "colorMutedLight": "",
          "colorMutedRegular": "",
          "colorRegular": "",
        },
        "rgb": Object {
          "colorDark": "",
          "colorLight": "",
          "colorMutedDark": "",
          "colorMutedLight": "",
          "colorMutedRegular": "",
          "colorRegular": "",
        },
      }
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  transition: background-color 1s;
  position: relative;
  min-height: 100vh;
  background-color: rgba(",
              [Function],
              ", 0.1);
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      theme={
        Object {
          "hex": Object {
            "colorDark": "",
            "colorLight": "",
            "colorMutedDark": "",
            "colorMutedLight": "",
            "colorMutedRegular": "",
            "colorRegular": "",
          },
          "rgb": Object {
            "colorDark": "",
            "colorLight": "",
            "colorMutedDark": "",
            "colorMutedLight": "",
            "colorMutedRegular": "",
            "colorRegular": "",
          },
        }
      }
    >
      <div
        className="c0"
      >
        Children
        <styled.div
          theme={
            Object {
              "hex": Object {
                "colorDark": "",
                "colorLight": "",
                "colorMutedDark": "",
                "colorMutedLight": "",
                "colorMutedRegular": "",
                "colorRegular": "",
              },
              "rgb": Object {
                "colorDark": "",
                "colorLight": "",
                "colorMutedDark": "",
                "colorMutedLight": "",
                "colorMutedRegular": "",
                "colorRegular": "",
              },
            }
          }
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  position: fixed;
  z-index: -1;
  top: 0;
  min-height: 100vh;
  min-width: 100vw;
  background: linear-gradient(
    0deg, rgba(",
                    [Function],
                    ", 0.1), rgba(",
                    [Function],
                    ", 0)
  );
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            theme={
              Object {
                "hex": Object {
                  "colorDark": "",
                  "colorLight": "",
                  "colorMutedDark": "",
                  "colorMutedLight": "",
                  "colorMutedRegular": "",
                  "colorRegular": "",
                },
                "rgb": Object {
                  "colorDark": "",
                  "colorLight": "",
                  "colorMutedDark": "",
                  "colorMutedLight": "",
                  "colorMutedRegular": "",
                  "colorRegular": "",
                },
              }
            }
          >
            <div
              className="c1"
            />
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Background>
`;
